examples/plot_gp_function.py

```diff
@@ -0,0 +1,183 @@
+    
+    
+    #print(utility_ei)
+    utility_function_poi = UtilityFunction(kind="poi", kappa = params['ucb']['kappa'], xi = params['poi']['xi'])
+    utility_poi = utility_function_poi.utility(x, optimizer3._gp, optimizer3._space.target.max())
+    
+    
+    # UCB
+    acq.plot(x, normalization(utility_ucb), label='UCB', color='purple')
+    acq.plot(x[np.argmax(utility_ucb)], np.max(normalization(utility_ucb)), '*', markersize=15, markerfacecolor='purple', markeredgecolor='k', markeredgewidth=1)
+    
+    # EI
+    acq.plot(x, normalization(utility_ei), label='EI', color='green')
+    acq.plot(x[np.argmax(utility_ei)], np.max(normalization(utility_ei)), '*', markersize=15, markerfacecolor='green', markeredgecolor='k', markeredgewidth=1)
+    
+    # Poi
+    acq.plot(x, normalization(utility_poi), label='POI', color='orange')
+    acq.plot(x[np.argmax(utility_poi)], np.max(normalization(utility_poi)), '*', markersize=15, markerfacecolor='orange', markeredgecolor='k', markeredgewidth=1)
+
+    
+    #acq.set_ylim((min(utility_ei) - 0.0005, max(utility_ei) + 0.0005))
+    acq.set_xlim((min(x), max(x)))
+    acq.set_ylim((-0.5,1.5)) #((min(min(utility_ucb), min(utility_ei), min(utility_poi)) - 0.5, max(max(utility_ucb), max(utility_ei), max(utility_poi)) + 0.5))
+    acq.set_ylabel('Utility', fontdict={'size':20})
+    acq.set_xlabel('x', fontdict={'size':20})
+    
+
+    
+    axis.legend(loc=2, bbox_to_anchor=(1.01, 1), borderaxespad=0.)
+    acq.legend(loc=2, bbox_to_anchor=(1.01, 1), borderaxespad=0.)
+    
+    plt.show()
+    
+    
+    
+def plot_convergence(optimizer1,optimizer2,optimizer3, x, target, params):
+    x = x.reshape(-1,1)
+    y = target(x)
+    
+
+    it=20
+    tar1=np.zeros(it)
+    tar2=np.zeros(it)
+    tar3=np.zeros(it)
+    point1=np.zeros(it)
+    point2=np.zeros(it)
+    point3=np.zeros(it)
+
+    utility_function1 = UtilityFunction(kind='ucb', kappa = params['ucb']['kappa'], xi = params['ucb']['xi'])
+    utility_function2 = UtilityFunction(kind='ei', kappa = params['ucb']['kappa'], xi = params['ei']['xi'])
+    utility_function3 = UtilityFunction(kind='poi', kappa = params['ucb']['kappa'], xi = params['poi']['xi'])    
+    
+    for i in range(it):
+
+        point1[i] = optimizer1.suggest(utility_function1)['x']
+        point2[i] = optimizer2.suggest(utility_function2)['x']
+        point3[i] = optimizer3.suggest(utility_function3)['x']
+        tar1[i]= target(point1[i])
+        tar2[i]= target(point2[i])
+        tar3[i]= target(point3[i])
+        optimizer1.register(params=point1[i], target=tar1[i])
+        optimizer2.register(params=point2[i], target=tar2[i])
+        optimizer3.register(params=point3[i], target=tar3[i])
+
+    
+    fig = plt.figure(figsize=(13, 6))
+    
+    acq = plt.subplot()
+    
+    fig.suptitle(
+        'Convergences to the optimum after 20 iterations',
```

Comment:
Avete il numero di iterazioni nella variabile `it`, usate quella invece che riscriverlo a mano. Tra l'altro metterei `it` come argomento della funzione, anche con default 20 per esempio